client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx

```diff
@@ -90,10 +91,11 @@ const DomainsStep: Step = function DomainsStep( { navigation, flow } ) {
 		shouldHideFreePlan = false
 	) => {
 		if ( ! suggestion ) {
+			// No domain was selected, 'decide later' button was clicked
 			setHideFreePlan( false );
 			setDomainCartItem( undefined );
 			setDomain( undefined );
-			return submit?.();
+			return submit?.( { freeDomain: true } );
```

Comment:
I am assuming when clicking on decide later is the same as free domain since a simple site(wordpress.com) is created anyway.
I might be wrong?